### PR TITLE
GEOPY-877:  Improve intersection detection in mask_by_extent

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-877 - Improve intersection detection in `mask_by_extent`**

- Remove orphan vertices on cell-based objects
- Add `inverse` option for the masking copy

### CODE REVIEW

File: geoh5py/objects/grid2d.py [hunk 1]
```diff
@@ -185,30 +186,35 @@ def copy_from_extent(
...
-                    origin[0] + delta_orig[0, 0],
-                    origin[1] + delta_orig[0, 1],
-                    origin[2],
-                ],
-                "u_count": np.sum(u_ind),
-                "v_count": np.sum(v_ind),
-            }
-        )
+        if not inverse:
+            assert self.u_cell_size is not None
```
Comment (benk-mira) [inline geoh5py/objects/grid2d.py:194]: raise ValueError?

**Review by benk-mira — CHANGES_REQUESTED** [review]
Looks good.  One small comment about assert vs. raise

File: geoh5py/objects/grid2d.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/objects/grid2d.py]: Good eye. Done

**Comment by domfournier** [comment]
@sebhmg Want to review this or can we go ahead with the release?

File: tests/clip_vertex_data_test.py
Comment (sebhmg) [inline tests/clip_vertex_data_test.py:1]: **suggestion (test):** add test with `inverse=True` on each type

File: tests/clip_vertex_data_test.py [hunk 2]
```diff
@@ -109,11 +112,42 @@ def test_clip_curve_data(tmp_path):
...
             clipped_pts = curve.copy_from_extent(parent=new_workspace, extent=extent)
-            clipped_d = clipped_pts.get_data("VertexValues")[0]
-            clipped_c = clipped_pts.get_data("CellValues")[0]
-            assert clipped_pts.n_vertices == clippings.sum()
-            assert np.all(clipped_d.values == data[0].values[clippings])
-            assert len(clipped_c.values) == clipped_pts.n_cells
+            assert clipped_pts is None
+
+
+def test_clip_surface(tmp_path):
```
Comment (sebhmg) [inline tests/clip_vertex_data_test.py:118]: **suggestion (test):** have a test for surface with `inverse=False`

**Review by sebhmg — CHANGES_REQUESTED** [review]
thanks!
preferably have test for each type with the new `inverse=True`

File: tests/clip_vertex_data_test.py
Comment (domfournier) [reply tests/clip_vertex_data_test.py:1]: Done

File: tests/clip_vertex_data_test.py (hunk 2 above)
Comment (domfournier) [reply tests/clip_vertex_data_test.py:124]: Done

**Review by sebhmg — APPROVED** [review]

**Pull request merged**